### PR TITLE
chore: 🤖 Constant Product AMM Implementation

### LINKED ISSUE #1290: [Idea]: Constant product Automated Maker Algorithm
### Idea Title

Constant product Automated Maker Algorithm  

### Idea Description

Add a production-ready implementation of the Constant Product AMM algorithm, similar to Uniswap V1's core mechanics. This implementation will provide foundational DEX (Decentralized Exchange) functionality with robust error handling, comprehensive testing, and proper mathematical precision.

### Potential Benefits

- [ ] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [ ] Improves code quality
- [X] Other

### Implementation Suggestions (Optional)

Token Operations
- Token creation with symbol, name, and decimals
- Liquidity position tracking

Pool Operations
- Pool initialization with initial liquidity
- Swap execution with fee collection
- Liquidity addition/removal

Price Calculations
- Spot price calculation
- Price impact estimation

### PR DESCRIPTION
**Pull request by yashksaini-coder**
✅ Closes: #1290

### Description
This PR raises a fix for the above mentioned issue to add the algorithm and approach for the Constant Product Automated market algorithm, with C code.

✅ Fixes #1290 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by yashksaini-coder** [comment]
@ajay-dhangar review this PR and assign it to me with proper labels

File: .vscode/settings.json
Comment (ajay-dhangar) [inline .vscode/settings.json:1]: don't push `.vscode/settings.json`

File: docs/Constant Product AMM/AMM.c
Comment (ajay-dhangar) [inline docs/Constant Product AMM/AMM.c:1]: we need only `.md` files not `docs/Constant Product AMM/AMM.c`

File: docs/Constant Product AMM/Cost Product AMM.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# 🤖 Simple Constant Product AMM Implementation
```
Comment (ajay-dhangar) [inline docs/Constant Product AMM/Cost Product AMM.md:12]: You need to follow our format and please carefully read our contribution guideline

**Comment by yashksaini-coder** [comment]
@ajay-dhangar what's the status of my PR

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Update your code based on our suggestion

**Review by ajay-dhangar — APPROVED** [review]
Done

**Comment by yashksaini-coder** [comment]
@ajay-dhangar can you add hacktoberfest labels too

**Pull request merged**